File: app/src/main/java/com/hana053/micropost/ui/ActivityModule.java
package com.hana053.micropost.ui;

import android.app.Activity;

import dagger.Module;
import dagger.Provides;
import lombok.AllArgsConstructor;
import rx.subscriptions.CompositeSubscription;

@Module
@AllArgsConstructor
@SuppressWarnings("WeakerAccess")
public class ActivityModule {

    private final Activity activity;

    @Provides
    @ActivityScope
    public Activity provideActivity() {
        return activity;
    }

    @Provides
    @ActivityScope
    public CompositeSubscription provideCompositeSubscription() {
        return new CompositeSubscription();
    }

    @Provides
    @ActivityScope
    public Navigator provideNavigator() {
        return new NavigatorImpl(activity);
    }


}

File: app/src/main/java/com/hana053/micropost/activity/ActivityModule.kt
package com.hana053.micropost.activity

import com.github.salomonbrys.kodein.Kodein
import com.github.salomonbrys.kodein.android.androidActivityScope
import com.github.salomonbrys.kodein.autoScopedSingleton
import com.github.salomonbrys.kodein.bind
import com.github.salomonbrys.kodein.instance

fun activityModule() = Kodein.Module {

    bind<Navigator>() with autoScopedSingleton(androidActivityScope) {
        NavigatorImpl(instance())
    }

}


